interfaces/C/ssids.f90

```diff
@@ -225,6 +232,90 @@ subroutine spral_ssids_analyse(ccheck, n, corder, cptr, crow, cval, cakeep, &
   call copy_inform_out(finform, cinform)
 end subroutine spral_ssids_analyse
 
+subroutine spral_ssids_analyse_topology(ccheck, n, order, ptr, row, val, cakeep, &
+   coptions, cinform, nregions, cregions) bind(C)
+use spral_ssids_ciface
+implicit none
+
+logical(C_BOOL), intent(in), value :: ccheck
+integer(C_INT), intent(in), value :: n, nregions
+integer(C_INT), intent(inout), dimension(n), optional :: order
+integer(C_INT64_T), intent(in), dimension(n+1) :: ptr
+type(C_PTR), intent(inout) :: cakeep
+type(spral_ssids_options), intent(in) :: coptions
+type(spral_ssids_inform), intent(out) :: cinform
+integer(C_INT), intent(in), dimension(ptr(n+1)-coptions%array_base) :: row
+real(C_DOUBLE), dimension(ptr(n+1)-coptions%array_base), optional :: val
+type(spral_numa_region), intent(in), dimension(nregions) :: cregions
+
+integer(C_INT64_T), dimension(:), allocatable :: ptr_find
```

Comment:
Perhaps slightly confusing naming, reads a lot like "to find" before you realize it is **f**ortran **ind**exed, but probably ok

Reply:
Yeah I was stuck for a better name

Reply:
Maybe `ptr_fidx` is clearer?